notebooks/Notebook1.md

```diff
@@ -2,33 +2,96 @@
+
+The language will have two overarcing setups. 1 build a roll roll {description} and then enacting stats on that roll. Rolls can be reused by a name and can be nested if possible. See design for more. The creation of stats would use the Scalapy package to grab some python graphing packages. There might be some scala graphing packages tho, so Ill keep an eye out for that. 
+
+Lastly, I will convert the scala to a single jar file as above. So the final checklist is 
+- Make the simplest working version of the DSL 
+- Make a Simple *working* GUI
+- Set up Jar adaptation 
+- Add Stats to DSL
+- Improve DSL 
+- Improve GUI 
+
+### Design :
+
+This is a preliminary design but this is what I want people to be able to create and explore with the system. 
+Making a roll: (basically a function that outputs a number) 
+```
+roll alpha {
+
+1d6 + 7
+
+}
+```
+note: 1d6 evaluates to a roll object. 
+
+Making a check: (basically a fucntion that outputs a succes or failure -- maybe add allowance for multiple successes reported?) 
+```
+check beta {
+
+5d6 if >10 
+
+}
+```
+note: 5d6 evaluates to a list of 1d6's that are then each rolled by if and compared to 10 
+
+Making a stat: (output stats with various parameters) 
+```
+roll gamma {
+ alpha + 1d6
+}
+
+stat beta (parameters)
+stat gamma (parameters)
+```
+This still needs some refinement. I want to break away from coding conventions, but we know thats very difficult. 
+
+
 
 ## Questions
 
 **What is the most pressing issue for your project? What design decision do
 you need to make, what implementation issue are you trying to solve, or how
 are you evaluating your design and implementation?**
 
+I feel much better about starting now. The most pressing issue is I have is I dont have experience with external implementations. Something my DSL relies on. Hopefully I can get guidence from the rest of the class on how that works. 
+I need to make the **big jump** from the *layout* and *construction* of a DSL to its actual design. I want to break away from the expectations of a traditional coding language and allow it to be relatively simple to write, especially since most applications will be simple. But the modularity and the recursiveness is defined well. 
+
 **What questions do you have for your critique partners? How can they best help
 you?**
 
+Does the structure make sense? Both the actual construction of the DSL (gui, scala etc) and the actual DSL design. Plus any name suggestions would be great! 
```

Comment:
I think the structure so far is excellent- it's very readable and seems fairly approachable to write. I'm excited to see how the GUI turns out, but I think your initial idea for the layout is pretty strong. Name-wise, I have no idea; some pun around die, dice, or roll might be good though.